utils/auth/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#[macro_use]
+extern crate lazy_static;
+
+use crypto::SignatureType;
+use jsonrpc_v2::Error as JsonRpcError;
+use jsonwebtoken::errors::Result as JWTResult;
+use jsonwebtoken::{decode, encode, DecodingKey, EncodingKey, Header, Validation};
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use wallet::KeyInfo;
+
+/// Error Enum for Authentification
+#[derive(Debug, Error, Serialize, Deserialize)]
+pub enum Error {
+    /// Filecoin Method does not exist
+    #[error("Filecoin method does not exist")]
+    MethodParam,
+    /// Invalid permissions to use specified method
+    #[error("Incorrect permissions to access method")]
+    InvalidPermissions,
+    /// Missing authentication header
+    #[error("Missing authentication header")]
+    NoAuthHeader,
+    #[error("{0}")]
+    Other(String),
+}
+
+lazy_static! {
+    /// Constants of all Levels of permissions
+    pub static ref ADMIN: Vec<String> = vec![
+        "read".to_string(),
+        "write".to_string(),
+        "sign".to_string(),
+        "admin".to_string()
+    ];
+    pub static ref SIGN: Vec<String> =
+        vec!["read".to_string(), "write".to_string(), "sign".to_string()];
+    pub static ref WRITE: Vec<String> = vec!["read".to_string(), "write".to_string()];
+    pub static ref READ: Vec<String> = vec!["read".to_string()];
+}
```

Comment:
Same here, none of this needs to be allocated, and why are they kept seperate

Reply:
yeah, they don't need to be separate, was just matching lotus